go-controller/pkg/ovn/policy_test.go

```diff
@@ -462,8 +462,8 @@ var _ = Describe("OVN NetworkPolicy Operations", func() {
 					},
 				)
 				nPodTest.populateLogicalSwitchCache(fakeOvn)
-				fakeOvn.controller.WatchPods()
 				fakeOvn.controller.WatchNamespaces()
+				fakeOvn.controller.WatchPods()
```

Comment:
why is this needed?

Reply:
Well, it did expose a bug :-). While unlikely, the namespace informer could certainly run after the pod informer...

Reply:
yeah I think there is a bug here, need to figure out where.

Reply:
The issue I think I found is that we can race exposed by the test code, when pod watcher is started before namespace. If the pod watcher starts first, locking the namespace will fail, causing addPodToNamespace to return nil (not error). The consequence here is that addLogicalPort finishes without error, but no address set is added and multicast is left unchecked. Afterwards, namespace watcher starts and AddNamespace will get called. Here we will get all pods in the namespace, and check to see if the pod has a pod.Status.PodIP. If it does not, then we will also miss adding the pod to address set and handling multicast enabled. I think there is a window of time here where the namespace could be added and the Status.PodIP is empty while CNI is finishing.

I think I've addressed this issue in this PR by making addPodToNamespace actually waitForNamespaceLocked. We need this anyway for other things in the code, so might as well do it one time here and ensure things are done correctly in the pod watcher.